src/dvc_data/hashfile/hash.py

```diff
@@ -190,5 +190,9 @@ def hash_file(
         assert ".dir" not in hash_info.value
         state.save(path, fs, hash_info)
 
-    meta = Meta(size=info["size"], isexec=is_exec(info.get("mode", 0)))
+    meta = Meta(
+        size=info["size"],
+        isexec=is_exec(info.get("mode", 0)),
+        version_id=info.get("version_id"),
```

Comment:
```suggestion
        version_id=info.get("VersionID"),
```
since it is coming from fs?

Reply:
Also probably need to capture it in state.